src/snowflake/snowpark/_internal/data_source/drivers/psycopg2_driver.py

```diff
@@ -0,0 +1,362 @@
+    Psycopg2TypeCode.CIDROID: StringType,
+    Psycopg2TypeCode.INETOID: StringType,
+    Psycopg2TypeCode.BPCHAROID: StringType,
+    Psycopg2TypeCode.VARCHAROID: StringType,
+    Psycopg2TypeCode.DATEOID: DateType,
+    Psycopg2TypeCode.TIMEOID: TimeType,
+    Psycopg2TypeCode.TIMESTAMPOID: TimestampType,
+    Psycopg2TypeCode.TIMESTAMPTZOID: TimestampType,
+    Psycopg2TypeCode.INTERVALOID: StringType,
+    Psycopg2TypeCode.TIMETZOID: TimeType,
+    Psycopg2TypeCode.BITOID: StringType,
+    Psycopg2TypeCode.VARBITOID: StringType,
+    Psycopg2TypeCode.NUMERICOID: DecimalType,
+    Psycopg2TypeCode.JSON: VariantType,
+    Psycopg2TypeCode.JSONB: VariantType,
+    Psycopg2TypeCode.MACADDR8: StringType,
+    Psycopg2TypeCode.UUID: StringType,
+    Psycopg2TypeCode.XML: StringType,
+    Psycopg2TypeCode.TSVECTOR: StringType,
+    Psycopg2TypeCode.TSQUERY: StringType,
+    Psycopg2TypeCode.TXID_SNAPSHOT: StringType,
+    Psycopg2TypeCode.PG_LSN: StringType,
+    Psycopg2TypeCode.PG_SNAPSHOT: StringType,
+}
+
+
+class Psycopg2Driver(BaseDriver):
+    def __init__(
+        self, create_connection: Callable[[], "Connection"], dbms_type: Enum
+    ) -> None:
+        super().__init__(create_connection, dbms_type)
+
+    def to_snow_type(self, schema: List[Any]) -> StructType:
+        # The psycopg2 spec is defined in the following links:
+        # https://www.psycopg.org/docs/cursor.html#cursor.description
+        # https://www.psycopg.org/docs/extensions.html#psycopg2.extensions.Column
+        fields = []
+        for (
+            name,
+            type_code,
+            _display_size,
+            _internal_size,
+            precision,
+            scale,
+            _null_ok,
+        ) in schema:
+            try:
+                type_code = Psycopg2TypeCode(type_code)
+            except ValueError:
+                raise NotImplementedError(
+                    f"Postgres type not supported: {type_code} for column: {name}"
+                )
+            snow_type = BASE_POSTGRES_TYPE_TO_SNOW_TYPE.get(type_code)
+            if snow_type is None:
+                raise NotImplementedError(
+                    f"Postgres type not supported: {type_code} for column: {name}"
+                )
+            if Psycopg2TypeCode(type_code) == Psycopg2TypeCode.NUMERICOID:
+                if not self.validate_numeric_precision_scale(precision, scale):
+                    logger.debug(
+                        f"Snowpark does not support column"
+                        f" {name} of type {type_code} with precision {precision} and scale {scale}. "
+                        "The default Numeric precision and scale will be used."
+                    )
+                    precision, scale = None, None
+                data_type = snow_type(
+                    precision if precision is not None else 38,
+                    scale if scale is not None else 0,
+                )
+            elif type_code == Psycopg2TypeCode.TIMESTAMPTZOID:
```

Comment:
just to confirm that another timestamp type is mapped to NTZ right?

Reply:
yes, the other one defaults to NTZ according to https://other-docs.snowflake.com/en/connectors/postgres6/view-data#postgresql-to-snowflake-data-type-mapping